# Data Wrangling for Final Project
## Import packages

In [1]:
import pandas as pd
import numpy as np

In [2]:
happy_vibes = pd.read_csv('happy_vibes.csv')
chill_vibes = pd.read_csv('chill_vibes.csv')
sad_vibes = pd.read_csv('sad_vibes.csv')
romance_vibes = pd.read_csv('romance_vibes.csv')
confidence_vibes = pd.read_csv('confidence_vibes.csv')

### Adding in column to label each mood

In [3]:
happy_vibes['Mood'] = 'Happy'
happy_vibes

Unnamed: 0                                               Name  \
0             0                                    I Ain't Worried   
1             1                                          As It Was   
2             2            Left and Right (Feat. Jung Kook of BTS)   
3             3                                               SNAP   
4             4                                            Sunroof   
..          ...                                                ...   
195          48                                               Sway   
196          49                                         Hey Lover!   
197          50                                        happy stuff   
198          52  Heartbeat (From the “Lyle, Lyle, Crocodile” Or...   
199          53                          What's Been On Your Mind?   

                   Track_ID  \
0    4h9wh7iOZ0GGn8QVp4RAOB   
1    4LRPiXqCikLlN15c3yImP7   
2    0mBP9X2gPCuapvpZ7TGDk3   
3    76OGwb5RA9h4FxQPT33ekc   
4    4h4QlmocP3IuwYEj2j14p8   
..                      ...   
195  2ajUl8lBLAXOXNpG4NEPMz   
196  7rC3P7tpWriaC4hYWKwGQd   
197  1C7eURT2UroBgmOEAX2eiR   
198  1gACe11pZiy8Xv3SY0ocyz   
199  3lpshfvmjrxGm9xr3x47F2   

                                                 Album              Artist  \
0    I Ain’t Worried (Music From The Motion Picture...         OneRepublic   
1                                            As It Was        Harry Styles   
2              Left and Right (Feat. Jung Kook of BTS)        Charlie Puth   
3                                                 SNAP           Rosa Linn   
4                                              Sunroof         Nicky Youre   
..                                                 ...                 ...   
195                                      Michael Bublé       Michael Bublé   
196                             Cyclones and Hey Lover               Wabie   
197                                        happy stuff  Walk Off the Earth   
198  Heartbeat (From the “Lyle, Lyle, Crocodile” Or...        Shawn Mendes   
199                          What's Been On Your Mind?            Surfaces   

                  Artist_ID  \
0    5Pwc4xIPtQLFEnJriah9YJ   
1    6KImCVD70vtIoJWnq6nGn3   
2    6VuMaDnrHyPL1p4EHjYLi7   
3    46xBNx0j6cwY6sD9LgMTm1   
4    7qmpXeNz2ojlMl2EEfkeLs   
..                      ...   
195  1GxkXlMwML1oSg5eLPiAz3   
196  3wOQaMNQU0rbzWyGXIudmE   
197  6jEiUoyyJNPHzSR0Nib6HX   
198  7n2wHs1TKAczGzO7Dd2rGr   
199  4ETSs924pXMzjIeD6E9b4u   

                                          Artist_Genre  Duration_ms  \
0                    ['piano rock', 'pop', 'pop rock']       148485   
1                                              ['pop']       167303   
2                    ['dance pop', 'pop', 'viral pop']       154486   
3                                                   []       179551   
4                                                   []       163025   
..                                                 ...          ...   
195  ['adult standards', 'canadian pop', 'jazz pop'...       188066   
196                                    ['bedroom pop']       162560   
197                      ['canadian rock', 'pop rock']       162134   
198  ['canadian pop', 'dance pop', 'pop', 'viral pop']       140443   
199                            ['bedroom soul', 'pop']       172490   

     Popularity  Explicit  ...  Loudness  Mode  Speechiness  Acousticness  \
0            97     False  ...    -5.927     1       0.0475       0.08260   
1            96     False  ...    -5.338     0       0.0557       0.34200   
2            93     False  ...    -4.898     1       0.0324       0.61900   
3            93     False  ...    -8.198     1       0.0638       0.10700   
4            93     False  ...    -5.110     1       0.0401       0.35200   
..          ...       ...  ...       ...   ...          ...           ...   
195          75     False  ...    -5.529     0       0.0313       0.75300   
196         

In [4]:
chill_vibes['Mood'] = 'Chill'
chill_vibes

Unnamed: 0                                      Name  \
0             0                       Because You Move Me   
1             1                      Escape (feat. Hayla)   
2             2                                   Rebirth   
3             3   The Last Goodbye (feat. Bettye LaVette)   
4             4                                     Salta   
..          ...                                       ...   
195          95                                    Sentir   
196          96  Gratitude - Sébastien Léger Remix (Edit)   
197          97                                   Ikebana   
198          98                                  Centauri   
199          99                                   Someone   

                   Track_ID                                    Album  \
0    05GvwwTLLID738BbKN1ze0                      Because You Move Me   
1    10oKSzRcwbZsog2uq2gb4b                     Escape (feat. Hayla)   
2    4Xz2mxHREzWiEr0AyCJuU6                      In Another Lifetime   
3    4XeIiGpUBshIfs9yrBDVZC  The Last Goodbye (feat. Bettye LaVette)   
4    5L8R2cyft2HfQ1IlfVAbcz                    Something, Everything   
..                      ...                                      ...   
195  5xb2vxUeGJ7Yeclvfaq1O8                           Temporary Love   
196  0m1YhId2LT8OMqQ7CnkzD6        Gratitude (Sébastien Léger Remix)   
197  4cYMt6WiUFShfZ6M1A9LPi                                  Ikebana   
198  6sbTqflJcUFfyAbiWC8def                                 Centauri   
199  01rAwlFDrRT2FzimFTXrKB                                  Someone   

               Artist               Artist_ID  \
0           Tinlicker  5EmEZjq8eHEC6qFnT63Lza   
1                 Kx5  2avRYQUWQpIkzJOEkf0MdY   
2           Tinlicker  5EmEZjq8eHEC6qFnT63Lza   
3              ODESZA  21mKp7DqtSNHhCAU2ugvUw   
4    Sultan + Shepard  14Tg9FvbNismPR1PJHxRau   
..                ...                     ...   
195          Mishegas  1F3BcbR6yzILOCzzA3i0Rh   
196    Above & Beyond  10gzBoINW3cLJfZUka8Zoe   
197    Nora Van Elken  04m3oUGzjO3EJTQidFzTgM   
198              Lstn  0OO1zgX3CUfJQwoOEBSKSp   
199             Nectr  3jJ4ftbuJR4j8F6DVjwt82   

                                          Artist_Genre  Duration_ms  \
0                                                   []       196375   
1                                                   []       240445   
2                                                   []       284563   
3                               ['chillwave', 'ninja']       358983   
4    ['canadian electronic', 'edm', 'progressive ho...       222250   
..                                                 ...          ...   
195                                     ['vapor soul']       140851   
196  ['edm', 'pop dance', 'progressive house', 'pro...       234098   
197                   ['deep house', 'tropical house']       169830   
198  ['chill beats', 'new french touch', 'tropical ...       210201   
199                                                 []       236149   

     Popularity  Explicit  ...  Loudness  Mode  Speechiness  Acousticness  \
0            76     False  ...    -9.506     0       0.1490       0.69100   
1            74     False  ...    -6.934     0       0.0719       0.17900   
2            63     False  ...    -7.152     1       0.0556       0.01540   
3            73     False  ...    -2.710     1       0.0773       0.00752   
4            63     False  ...   -10.177     0       0.0327       0.04480   
..          ...       ...  ...       ...   ...          ...           ...   
195          47     False  ...    -9.155     1       0.0412       0.37700   
196          47     False  ...    -6.675     0       0.0454       0.00410   
197          47     False  ...    -5.446     0       0.0266       0.05570   
198          47     False  ...   -10.371     0       0.0396       0.19600   
199          47     False  ...    -9.106     1       0.0407       0.34500   

     Instrumentalness  Liveness  Valence

In [5]:
sad_vibes['Mood'] = 'Sad'
sad_vibes

Unnamed: 0                                               Name  \
0             0                                       Another Love   
1             1                               lovely (with Khalid)   
2             2                               I'm Not The Only One   
3             3  Running Up That Hill (A Deal With God) - 2018 ...   
4             4                                            Heather   
..          ...                                                ...   
195         195          Why Don't You Love Me (feat. Demi Lovato)   
196         196                              Don't Forget About Me   
197         197                                           Champion   
198         198                                   The Hardest Part   
199         199                                   Heart Like Yours   

                   Track_ID  \
0    3JvKfv6T31zO0ini8iNItO   
1    0u2P5u6lvoDfwTYjAADbn4   
2    7795WJLVKJoAyVoOtCWqXN   
3    29d0nY7TzCoi22XBqDQkiP   
4    4xqrdfXkTW4T0RauPLv3WA   
..                      ...   
195  0cEBv7oJxFp3bfTFqWFI0R   
196  4gVZhkcsZcwQ4mU1gEeXEO   
197  1z1mMZtZW0gvR9FSPc7JpJ   
198  1ODRbz1jZsSrTkpTVcYElX   
199  1ym2ZnNbIV6TB4YUjAl2Z5   

                                                 Album           Artist  \
0                               Long Way Down (Deluxe)        Tom Odell   
1                                 lovely (with Khalid)    Billie Eilish   
2                                   In The Lonely Hour        Sam Smith   
3                       Hounds of Love (2018 Remaster)        Kate Bush   
4                                             Kid Krow       Conan Gray   
..                                                 ...              ...   
195                                           Whatever   Hot Chelle Rae   
196                                               XIII           CLOVES   
197                                           Champion            Elina   
198                                         Aftershock     Alexander 23   
199  If I Stay (Original Motion Picture Soundtrack)...  Various Artists   

                  Artist_ID  \
0    2txHhyCwHjUEpJjWrEyqyX   
1    6qqNVTkY8uBg9cP3Jd7DAH   
2    2wY79sveU1sp5g7SokKOiI   
3    1aSxMhuvixZ8h9dK9jIDwL   
4    4Uc8Dsxct0oMqx0P6i60ea   
..                      ...   
195  6jTnHxhb6cDCaCu4rdvsQ0   
196  355SqtHY4qKt2wIXrWku0c   
197  4K9OTkRXEFL6NDXFTqVmq9   
198  6sFHvCyqklnJpXC9Nh1aag   
199  0LyfQWJT6nXafLPZqxe9Of   

                                          Artist_Genre  Duration_ms  \
0                                 ['chill pop', 'pop']       244360   
1                     ['art pop', 'electropop', 'pop']       200185   
2                       ['dance pop', 'pop', 'uk pop']       239316   
3    ['art pop', 'art rock', 'baroque pop', 'new wa...       300840   
4                               ['bedroom pop', 'pop']       198040   
..                                                 ...          ...   
195  ['australian pop', 'dance pop', 'dance rock', ...       211413   
196                     ['australian alternative pop']       268853   
197                 ['alt z', 'australian pop', 'pop']       173237   
198                         ['alt z', 'pop', 'uk pop']       203600   
199                                                 []       200560   

     Popularity  Explicit  ...  Loudness  Mode  Speechiness  Acousticness  \
0            93      True  ...    -8.532     0       0.0400         0.695   
1            89     False  ...   -10.109     0       0.0333         0.934   
2            88     False  ...    -5.795     1       0.0361         0.529   
3            87     False  ...   -11.903     0       0.0596         0.659   
4            87     False  ...    -7.301     1       0.0333         0.584   
..          ...       ...  ...       ...   ...          ...           ...   
195          61     False  ...    -6.494     1       0.0362         0.121   
196          61     False  ...    -9.083     0  

In [6]:
romance_vibes['Mood'] = 'Romance'
romance_vibes

Unnamed: 0                                       Name  \
0             0                                 Heat Waves   
1             1                       You're Somebody Else   
2             2                              Call Me Lover   
3             3                         You Get Me So High   
4             4                                    I Found   
..          ...                                        ...   
195          21                      Good & Plenty - Remix   
196          22            Freak Like Me - Spotify Singles   
197          23                  On The Way (feat. Mila J)   
198          24  Splash (feat. Jhené Aiko & Ty Dolla $ign)   
199          25                                      Dream   

                   Track_ID                                            Album  \
0    6CDzDgIUqeDY5g8ujExx2f                                       Heat Waves   
1    0pdKRp2sUthTPe7RLWpPqQ                             You're Somebody Else   
2    4Pg6x9W2ziSJh5iym9XRTB                              Hypersonic Missiles   
3    7zwn1eykZtZ5LODrf7c0tS  Hard To Imagine The Neighbourhood Ever Changing   
4    4lRuRtvftdSSUA7hLP7zAr                                         Pilot EP   
..                      ...                                              ...   
195  6T3Ebo7EOh8cUOyE4OhFpp                            Good & Plenty (Remix)   
196  7lxF9OOlLhxslC3BEDIMfB                                  Spotify Singles   
197  423NhOTzoqgeaIJbrUzSoo                                Chilombo (Deluxe)   
198  5Ip2VxpYt1dCD7MKmjrw52                                           LEGEND   
199  1mAQAh6yg1dTmnlE84ZvEo                                           Breezy   

                Artist               Artist_ID  \
0        Glass Animals  4yvcSjfu4PC0CYQyLy4wSq   
1           flora cash  6GpcBKNmZDIrRzYkPJu7Wd   
2           Sam Fender  6zlR5ttMfMNmwf2lecU9Cc   
3    The Neighbourhood  77SW9BnxLY8rJ0RciFqkHh   
4            Amber Run  0MmnmsAuQKRFpo6vJElcaU   
..                 ...                     ...   
195         Lucky Daye  5Vuvs6Py2JRU7WiFDVsI7J   
196              Chlöe  1FtBEIWAwvw5ymBen5GICR   
197         Jhené Aiko  5ZS223C6JyBfXasXxrRqOk   
198        John Legend  5y2Xq6xcjJb2jVM54GHK3t   
199        Chris Brown  7bXgB6jMjp9ATFy66eO08Z   

                                          Artist_Genre  Duration_ms  \
0          ['gauze pop', 'indietronica', 'shiver pop']       238805   
1     ['indie pop', 'scandipop', 'swedish indie folk']       218883   
2          ['modern rock', 'north east england indie']       201106   
3    ['modern alternative rock', 'modern rock', 'po...       153000   
4                                       ['indie folk']       274226   
..                                                 ...          ...   
195                                     ['pop', 'r&b']       204705   
196             ['pop', 'post-teen pop', 'r&b', 'rap']       188989   
197               ['pop', 'r&b', 'urban contemporary']       222600   
198  ['neo mellow', 'neo soul', 'pop', 'pop soul', ...       234498   
199                        ['dance pop', 'pop', 'r&b']       190440   

     Popularity  Explicit  ...  Loudness  Mode  Speechiness  Acousticness  \
0            85     False  ...    -6.900     1       0.0944         0.440   
1            75     False  ...    -9.638     1       0.0295         0.841   
2            60     False  ...    -4.724     0       0.0376         0.177   
3            83     False  ...    -6.099     0       0.0542         0.186   
4            60     False  ...   -12.433     0       0.0474         0.879   
..          ...       ...  ...       ...   ...          ...           ...   
195          65     False  ...    -6.543     0       0.1070         0.811   
196          65     False  ...   -11.152     0       0.2310         0.232   
197          64      True  ...   -11.626     0       0.2800         0.756   
198          63      True  ...    -6.271     0       0.0557         0.310  

In [7]:
confidence_vibes['Mood'] = 'Confidence'
confidence_vibes

Unnamed: 0                               Name                Track_ID  \
0             0                  Super Freaky Girl  4C6Uex2ILwJi9sZXRdmqXp   
1             1                                Her  5hHvFrMnwic82WwRLr4WY0   
2             2          Unholy (feat. Kim Petras)  3nqQXoyQOWXiESFLlDF1hG   
3             3                            BIG MAD  0bspC5fLlWgwCM6Rx1YUIh   
4             4          Tomorrow 2 (with Cardi B)  7KXVIAuw3m2hxZanrpSXS3   
..          ...                                ...                     ...   
155          45                                Oh!  4e72LGraX8sW5iztJvS5fo   
156          46                 God Complex (Mojo)  1B3ouXA9iRkqY1ywvZbzZ5   
157          47                            I EXIST  3eI62ZWWxkl4LUaBqXvA6Y   
158          48  Apartment Complex on the Eastside  1cRk9clJhi67mcfTxNfnjU   
159          49                Black Roller Coming  5OLFlnmmC8gQHiFisYvrAh   

                            Album               Artist  \
0               Super Freaky Girl          Nicki Minaj   
1                      Traumazine  Megan Thee Stallion   
2       Unholy (feat. Kim Petras)            Sam Smith   
3                         BIG MAD                Ktlyn   
4       Tomorrow 2 (with Cardi B)             GloRilla   
..                            ...                  ...   
155                           Oh!     The Linda Lindas   
156            God Complex (Mojo)            poutyface   
157                       I EXIST               VUKOVI   
158  It Took A Lot To Get To This         Dylan Fraser   
159                 Spirit Rising         Philip Sayce   

                  Artist_ID  \
0    0hCNtLu0JehylgoiP8L4Gh   
1    181bsRPaVXVlUKXrxwZfHK   
2    2wY79sveU1sp5g7SokKOiI   
3    6FaLegXtqjGZBH5VFrOlwG   
4    2qoQgPAilErOKCwE2Y8wOG   
..                      ...   
155  13dTrWNNrnZ3AkgNyQNKP5   
156  0H44O4IYqpTOGx4c5nV37f   
157  1844Ua6R4gOuH6GLdlR4dt   
158  6Awp6fgyzqQ2XuEvOMjq8D   
159  5Npr4HpRE8YlsisRjN9T8h   

                                          Artist_Genre  Duration_ms  \
0    ['dance pop', 'hip pop', 'pop', 'queens hip ho...       170977   
1    ['dance pop', 'houston rap', 'pop', 'rap', 'tr...       137107   
2                       ['dance pop', 'pop', 'uk pop']       156943   
3                                                   []       143111   
4                                                   []       209811   
..                                                 ...          ...   
155                                     ['riot grrrl']       155821   
156         ['alt z', 'modern indie pop', 'viral rap']       106740   
157  ['modern alternative rock', 'modern rock', 'pi...       269666   
158                                                 []       247545   
159  ['blues rock', 'modern blues', 'modern blues r...       169287   

     Popularity  Explicit  ...  Loudness  Mode  Speechiness  Acousticness  \
0            93      True  ...    -2.653     1       0.2410      0.064500   
1            78      True  ...    -4.760     1       0.3260      0.374000   
2            97     False  ...    -7.375     1       0.0864      0.013000   
3            69      True  ...    -6.100     1       0.0591      0.029300   
4            77      True  ...    -3.930     1       0.4060      0.069500   
..          ...       ...  ...       ...   ...          ...           ...   
155          48     False  ...    -3.313     1       0.0609      0.000039   
156          48      True  ...    -3.333     0       0.0571      0.004390   
157          49      True  ...    -4.702     1       0.0680      0.000063   
158          41      True  ...    -5.433     1       0.0577      0.075800   
159          44     False  ...    -5.443     0       0.0479      0.009820   

     Instrumentalness  Liveness  Valence    Tempo            Playlist  \
0            0.000018    0.3090    0.912  133.010                 BBE   
1            0.000033    0.1010    0.819  119.918                 BBE   
2

### Combining individual datasets together.

In [8]:
master_mood = pd.concat([happy_vibes, chill_vibes, sad_vibes, romance_vibes, confidence_vibes])
master_mood

Unnamed: 0                                     Name  \
0             0                          I Ain't Worried   
1             1                                As It Was   
2             2  Left and Right (Feat. Jung Kook of BTS)   
3             3                                     SNAP   
4             4                                  Sunroof   
..          ...                                      ...   
155          45                                      Oh!   
156          46                       God Complex (Mojo)   
157          47                                  I EXIST   
158          48        Apartment Complex on the Eastside   
159          49                      Black Roller Coming   

                   Track_ID  \
0    4h9wh7iOZ0GGn8QVp4RAOB   
1    4LRPiXqCikLlN15c3yImP7   
2    0mBP9X2gPCuapvpZ7TGDk3   
3    76OGwb5RA9h4FxQPT33ekc   
4    4h4QlmocP3IuwYEj2j14p8   
..                      ...   
155  4e72LGraX8sW5iztJvS5fo   
156  1B3ouXA9iRkqY1ywvZbzZ5   
157  3eI62ZWWxkl4LUaBqXvA6Y   
158  1cRk9clJhi67mcfTxNfnjU   
159  5OLFlnmmC8gQHiFisYvrAh   

                                                 Album            Artist  \
0    I Ain’t Worried (Music From The Motion Picture...       OneRepublic   
1                                            As It Was      Harry Styles   
2              Left and Right (Feat. Jung Kook of BTS)      Charlie Puth   
3                                                 SNAP         Rosa Linn   
4                                              Sunroof       Nicky Youre   
..                                                 ...               ...   
155                                                Oh!  The Linda Lindas   
156                                 God Complex (Mojo)         poutyface   
157                                            I EXIST            VUKOVI   
158                       It Took A Lot To Get To This      Dylan Fraser   
159                                      Spirit Rising      Philip Sayce   

                  Artist_ID  \
0    5Pwc4xIPtQLFEnJriah9YJ   
1    6KImCVD70vtIoJWnq6nGn3   
2    6VuMaDnrHyPL1p4EHjYLi7   
3    46xBNx0j6cwY6sD9LgMTm1   
4    7qmpXeNz2ojlMl2EEfkeLs   
..                      ...   
155  13dTrWNNrnZ3AkgNyQNKP5   
156  0H44O4IYqpTOGx4c5nV37f   
157  1844Ua6R4gOuH6GLdlR4dt   
158  6Awp6fgyzqQ2XuEvOMjq8D   
159  5Npr4HpRE8YlsisRjN9T8h   

                                          Artist_Genre  Duration_ms  \
0                    ['piano rock', 'pop', 'pop rock']       148485   
1                                              ['pop']       167303   
2                    ['dance pop', 'pop', 'viral pop']       154486   
3                                                   []       179551   
4                                                   []       163025   
..                                                 ...          ...   
155                                     ['riot grrrl']       155821   
156         ['alt z', 'modern indie pop', 'viral rap']       106740   
157  ['modern alternative rock', 'modern rock', 'pi...       269666   
158                                                 []       247545   
159  ['blues rock', 'modern blues', 'modern blues r...       169287   

     Popularity  Explicit  ...  Loudness  Mode  Speechiness  Acousticness  \
0            97     False  ...    -5.927     1       0.0475      0.082600   
1            96     False  ...    -5.338     0       0.0557      0.342000   
2            93     False  ...    -4.898     1       0.0324      0.619000   
3            93     False  ...    -8.198     1       0.0638      0.107000   
4            93     False  ...    -5.110     1       0.0401      0.352000   
..          ...       ...  ...       ...   ...          ...           ...   
155          48     False  ...    -3.313     1       0.0609      0.000039   
156          48      True  ...    -3.333     0       0.0571      0.004390   
157          49      True  ...    -4.702     1       0.0680      0.000063   
15

### Reset index

In [9]:
master_mood1 = master_mood.reset_index()
master_mood1

index  Unnamed: 0                                     Name  \
0        0           0                          I Ain't Worried   
1        1           1                                As It Was   
2        2           2  Left and Right (Feat. Jung Kook of BTS)   
3        3           3                                     SNAP   
4        4           4                                  Sunroof   
..     ...         ...                                      ...   
955    155          45                                      Oh!   
956    156          46                       God Complex (Mojo)   
957    157          47                                  I EXIST   
958    158          48        Apartment Complex on the Eastside   
959    159          49                      Black Roller Coming   

                   Track_ID  \
0    4h9wh7iOZ0GGn8QVp4RAOB   
1    4LRPiXqCikLlN15c3yImP7   
2    0mBP9X2gPCuapvpZ7TGDk3   
3    76OGwb5RA9h4FxQPT33ekc   
4    4h4QlmocP3IuwYEj2j14p8   
..                      ...   
955  4e72LGraX8sW5iztJvS5fo   
956  1B3ouXA9iRkqY1ywvZbzZ5   
957  3eI62ZWWxkl4LUaBqXvA6Y   
958  1cRk9clJhi67mcfTxNfnjU   
959  5OLFlnmmC8gQHiFisYvrAh   

                                                 Album            Artist  \
0    I Ain’t Worried (Music From The Motion Picture...       OneRepublic   
1                                            As It Was      Harry Styles   
2              Left and Right (Feat. Jung Kook of BTS)      Charlie Puth   
3                                                 SNAP         Rosa Linn   
4                                              Sunroof       Nicky Youre   
..                                                 ...               ...   
955                                                Oh!  The Linda Lindas   
956                                 God Complex (Mojo)         poutyface   
957                                            I EXIST            VUKOVI   
958                       It Took A Lot To Get To This      Dylan Fraser   
959                                      Spirit Rising      Philip Sayce   

                  Artist_ID  \
0    5Pwc4xIPtQLFEnJriah9YJ   
1    6KImCVD70vtIoJWnq6nGn3   
2    6VuMaDnrHyPL1p4EHjYLi7   
3    46xBNx0j6cwY6sD9LgMTm1   
4    7qmpXeNz2ojlMl2EEfkeLs   
..                      ...   
955  13dTrWNNrnZ3AkgNyQNKP5   
956  0H44O4IYqpTOGx4c5nV37f   
957  1844Ua6R4gOuH6GLdlR4dt   
958  6Awp6fgyzqQ2XuEvOMjq8D   
959  5Npr4HpRE8YlsisRjN9T8h   

                                          Artist_Genre  Duration_ms  \
0                    ['piano rock', 'pop', 'pop rock']       148485   
1                                              ['pop']       167303   
2                    ['dance pop', 'pop', 'viral pop']       154486   
3                                                   []       179551   
4                                                   []       163025   
..                                                 ...          ...   
955                                     ['riot grrrl']       155821   
956         ['alt z', 'modern indie pop', 'viral rap']       106740   
957  ['modern alternative rock', 'modern rock', 'pi...       269666   
958                                                 []       247545   
959  ['blues rock', 'modern blues', 'modern blues r...       169287   

     Popularity  ...  Loudness  Mode  Speechiness  Acousticness  \
0            97  ...    -5.927     1       0.0475      0.082600   
1            96  ...    -5.338     0       0.0557      0.342000   
2            93  ...    -4.898     1       0.0324      0.619000   
3            93  ...    -8.198     1       0.0638      0.107000   
4            93  ...    -5.110     1       0.0401      0.352000   
..          ...  ...       ...   ...          ...           ...   
955          48  ...    -3.313     1       0.0609      0.000039   
956          48  ...    -3.333     0       0.0571      0.004390   
957          49  ...    -4.702     1       0.0680      0.000063   
958          41  .

### Removing columns that contain old indexes

In [10]:
master_mood2 = master_mood1.drop(['Unnamed: 0', 'index'], axis = 1)
master_mood2

Name                Track_ID  \
0                            I Ain't Worried  4h9wh7iOZ0GGn8QVp4RAOB   
1                                  As It Was  4LRPiXqCikLlN15c3yImP7   
2    Left and Right (Feat. Jung Kook of BTS)  0mBP9X2gPCuapvpZ7TGDk3   
3                                       SNAP  76OGwb5RA9h4FxQPT33ekc   
4                                    Sunroof  4h4QlmocP3IuwYEj2j14p8   
..                                       ...                     ...   
955                                      Oh!  4e72LGraX8sW5iztJvS5fo   
956                       God Complex (Mojo)  1B3ouXA9iRkqY1ywvZbzZ5   
957                                  I EXIST  3eI62ZWWxkl4LUaBqXvA6Y   
958        Apartment Complex on the Eastside  1cRk9clJhi67mcfTxNfnjU   
959                      Black Roller Coming  5OLFlnmmC8gQHiFisYvrAh   

                                                 Album            Artist  \
0    I Ain’t Worried (Music From The Motion Picture...       OneRepublic   
1                                            As It Was      Harry Styles   
2              Left and Right (Feat. Jung Kook of BTS)      Charlie Puth   
3                                                 SNAP         Rosa Linn   
4                                              Sunroof       Nicky Youre   
..                                                 ...               ...   
955                                                Oh!  The Linda Lindas   
956                                 God Complex (Mojo)         poutyface   
957                                            I EXIST            VUKOVI   
958                       It Took A Lot To Get To This      Dylan Fraser   
959                                      Spirit Rising      Philip Sayce   

                  Artist_ID  \
0    5Pwc4xIPtQLFEnJriah9YJ   
1    6KImCVD70vtIoJWnq6nGn3   
2    6VuMaDnrHyPL1p4EHjYLi7   
3    46xBNx0j6cwY6sD9LgMTm1   
4    7qmpXeNz2ojlMl2EEfkeLs   
..                      ...   
955  13dTrWNNrnZ3AkgNyQNKP5   
956  0H44O4IYqpTOGx4c5nV37f   
957  1844Ua6R4gOuH6GLdlR4dt   
958  6Awp6fgyzqQ2XuEvOMjq8D   
959  5Npr4HpRE8YlsisRjN9T8h   

                                          Artist_Genre  Duration_ms  \
0                    ['piano rock', 'pop', 'pop rock']       148485   
1                                              ['pop']       167303   
2                    ['dance pop', 'pop', 'viral pop']       154486   
3                                                   []       179551   
4                                                   []       163025   
..                                                 ...          ...   
955                                     ['riot grrrl']       155821   
956         ['alt z', 'modern indie pop', 'viral rap']       106740   
957  ['modern alternative rock', 'modern rock', 'pi...       269666   
958                                                 []       247545   
959  ['blues rock', 'modern blues', 'modern blues r...       169287   

     Popularity  Explicit  Danceability  ...  Loudness  Mode  Speechiness  \
0            97     False         0.704  ...    -5.927     1       0.0475   
1            96     False         0.520  ...    -5.338     0       0.0557   
2            93     False         0.881  ...    -4.898     1       0.0324   
3            93     False         0.565  ...    -8.198     1       0.0638   
4            93     False         0.768  ...    -5.110     1       0.0401   
..          ...       ...           ...  ...       ...   ...          ...   
955          48     False         0.639  ...    -3.313     1       0.0609   
956          48      True         0.573  ...    -3.333     0       0.0571   
957          49      True         0.469  ...    -4.702     1       0.0680   
958          41      True         0.469  ...    -5.433     1       0.0577   
959          44     False         0.400  ...    -5.443     0       0.0479   

     Acousticness  Instrumentalness  Liveness  Valence    Tempo  \
0        0.082600          0.000745    0.054

### Look at songs that are listed in multiple categories (assumption of independence)

In [11]:
dup_songs = master_mood2.loc[master_mood1.duplicated(subset = ['Name'], keep = False), :]

In [12]:
dup_songs.sort_values(by='Name')

Name  \
303                                          Afterglow   
230                                          Afterglow   
842                                         Big Energy   
75                                          Big Energy   
352                                            Breathe   
388                                            Breathe   
710                         Can't Help Falling in Love   
450                         Can't Help Falling in Love   
799                                              Dream   
612                                              Dream   
83                              Dreams - 2004 Remaster   
705                             Dreams - 2004 Remaster   
249                                               Eyes   
765                                               Eyes   
830                    Freak Like Me - Spotify Singles   
796                    Freak Like Me - Spotify Singles   
255                                          Gratitude   
381                                          Gratitude   
172                                         Heat Waves   
12                                          Heat Waves   
600                                         Heat Waves   
675                                            Hold On   
791                                            Hold On   
552                    I Will Follow You into the Dark   
731                    I Will Follow You into the Dark   
433                                           Memories   
408                                           Memories   
891                              Persuasive (with SZA)   
193                              Persuasive (with SZA)   
115                                Put Your Records On   
41                                 Put Your Records On   
593                                                Run   
216                                                Run   
403  Running Up That Hill (A Deal With God) - 2018 ...   
77   Running Up That Hill (A Deal With God) - 2018 ...   
481                                        Skinny Love   
490                                        Skinny Love   
953                                              Smile   
279                                              Smile   
735                                              Smile   
176                                           Sunshine   
25                                            Sunshine   
619                                  Until I Found You   
7                                    Until I Found You   
624                                           sheluvme   
182                                           sheluvme   

                   Track_ID                           Album  \
303  4Ir2UxDlAm01qmv3eSZYlP                       Afterglow   
230  2Ftoh82TZBdd5VduEm8T44                       Afterglow   
842  6Zu3aw7FfjAF9WA0fA81Oq                      Big Energy   
75   5lHxARJJrZc3ss4ZVmGWfz                      Big Energy   
352  5ag0L0xkUGJrkXmalqOfho                         Breathe   
388  1rx0RK6hNBRFKAjeNrkVNn                         Breathe   
710  44AyOl4qVkzS48vBsbNXaC                     Blue Hawaii   
450  6yIHGmQLJxWAUZ1ZkENemN                          Better   
799  1mAQAh6yg1dTmnlE84ZvEo                          Breezy   
612  25DpvTS4zsmLiVLx8Zv3N0                 Smoke + Mirrors   
83   0ofHAoxe9vBkTCp2UQIavz          Rumours (Super Deluxe)   
705  0ofHAoxe9vBkTCp2UQIavz          Rumours (Super Deluxe)   
249  7uEYsZQ6fuvvbsnGVAp8Ls                            Eyes   
765  1a9czdo8IX5InOiSOg6xpv                            Eyes   
830  7lxF9OOlLhxslC3BEDIMfB                 Spotify Singles   
796  7lxF9OOlLhxslC3BEDIMfB                 Spotify Singles   
255  5CvaeEX9dU9ZhwyoVIIgTO                       Gratitude   
381  3fY7n6r6Ky1OuFyf0qgE1O                       Gratitude   
172  6CDzDgIUqeDY5g8ujExx2f                      Heat Waves   
12   02MWAaffLxlfxAUY7c5dvx      Dreamland (+ Bonus Levels)   


### There were only 9 duplicate songs. I went through and listened to them, and I am going to keep them in the category I feel they fit best in based on lyrical contents. 

In [13]:
master_no_dupes = master_mood2.drop([705, 830, 172, 12, 152,193, 77,7, 182])
master_no_dupes

Name                Track_ID  \
0                            I Ain't Worried  4h9wh7iOZ0GGn8QVp4RAOB   
1                                  As It Was  4LRPiXqCikLlN15c3yImP7   
2    Left and Right (Feat. Jung Kook of BTS)  0mBP9X2gPCuapvpZ7TGDk3   
3                                       SNAP  76OGwb5RA9h4FxQPT33ekc   
4                                    Sunroof  4h4QlmocP3IuwYEj2j14p8   
..                                       ...                     ...   
955                                      Oh!  4e72LGraX8sW5iztJvS5fo   
956                       God Complex (Mojo)  1B3ouXA9iRkqY1ywvZbzZ5   
957                                  I EXIST  3eI62ZWWxkl4LUaBqXvA6Y   
958        Apartment Complex on the Eastside  1cRk9clJhi67mcfTxNfnjU   
959                      Black Roller Coming  5OLFlnmmC8gQHiFisYvrAh   

                                                 Album            Artist  \
0    I Ain’t Worried (Music From The Motion Picture...       OneRepublic   
1                                            As It Was      Harry Styles   
2              Left and Right (Feat. Jung Kook of BTS)      Charlie Puth   
3                                                 SNAP         Rosa Linn   
4                                              Sunroof       Nicky Youre   
..                                                 ...               ...   
955                                                Oh!  The Linda Lindas   
956                                 God Complex (Mojo)         poutyface   
957                                            I EXIST            VUKOVI   
958                       It Took A Lot To Get To This      Dylan Fraser   
959                                      Spirit Rising      Philip Sayce   

                  Artist_ID  \
0    5Pwc4xIPtQLFEnJriah9YJ   
1    6KImCVD70vtIoJWnq6nGn3   
2    6VuMaDnrHyPL1p4EHjYLi7   
3    46xBNx0j6cwY6sD9LgMTm1   
4    7qmpXeNz2ojlMl2EEfkeLs   
..                      ...   
955  13dTrWNNrnZ3AkgNyQNKP5   
956  0H44O4IYqpTOGx4c5nV37f   
957  1844Ua6R4gOuH6GLdlR4dt   
958  6Awp6fgyzqQ2XuEvOMjq8D   
959  5Npr4HpRE8YlsisRjN9T8h   

                                          Artist_Genre  Duration_ms  \
0                    ['piano rock', 'pop', 'pop rock']       148485   
1                                              ['pop']       167303   
2                    ['dance pop', 'pop', 'viral pop']       154486   
3                                                   []       179551   
4                                                   []       163025   
..                                                 ...          ...   
955                                     ['riot grrrl']       155821   
956         ['alt z', 'modern indie pop', 'viral rap']       106740   
957  ['modern alternative rock', 'modern rock', 'pi...       269666   
958                                                 []       247545   
959  ['blues rock', 'modern blues', 'modern blues r...       169287   

     Popularity  Explicit  Danceability  ...  Loudness  Mode  Speechiness  \
0            97     False         0.704  ...    -5.927     1       0.0475   
1            96     False         0.520  ...    -5.338     0       0.0557   
2            93     False         0.881  ...    -4.898     1       0.0324   
3            93     False         0.565  ...    -8.198     1       0.0638   
4            93     False         0.768  ...    -5.110     1       0.0401   
..          ...       ...           ...  ...       ...   ...          ...   
955          48     False         0.639  ...    -3.313     1       0.0609   
956          48      True         0.573  ...    -3.333     0       0.0571   
957          49      True         0.469  ...    -4.702     1       0.0680   
958          41      True         0.469  ...    -5.433     1       0.0577   
959          44     False         0.400  ...    -5.443     0       0.0479   

     Acousticness  Instrumentalness  Liveness  Valence    Tempo  \
0        0.082600          0.000745    0.054

### Recoding Mood Genres 

In [14]:
def mood_recode(mood):
    if mood == 'Happy':
        return 1
    if mood == 'Chill':
        return 2
    if mood == 'Sad':
        return 3
    if mood == 'Romance':
        return 4
    if mood == 'Confidence':
        return 5

In [15]:
master_no_dupes['MoodR']= master_no_dupes['Mood'].apply(mood_recode)
master_no_dupes

Name                Track_ID  \
0                            I Ain't Worried  4h9wh7iOZ0GGn8QVp4RAOB   
1                                  As It Was  4LRPiXqCikLlN15c3yImP7   
2    Left and Right (Feat. Jung Kook of BTS)  0mBP9X2gPCuapvpZ7TGDk3   
3                                       SNAP  76OGwb5RA9h4FxQPT33ekc   
4                                    Sunroof  4h4QlmocP3IuwYEj2j14p8   
..                                       ...                     ...   
955                                      Oh!  4e72LGraX8sW5iztJvS5fo   
956                       God Complex (Mojo)  1B3ouXA9iRkqY1ywvZbzZ5   
957                                  I EXIST  3eI62ZWWxkl4LUaBqXvA6Y   
958        Apartment Complex on the Eastside  1cRk9clJhi67mcfTxNfnjU   
959                      Black Roller Coming  5OLFlnmmC8gQHiFisYvrAh   

                                                 Album            Artist  \
0    I Ain’t Worried (Music From The Motion Picture...       OneRepublic   
1                                            As It Was      Harry Styles   
2              Left and Right (Feat. Jung Kook of BTS)      Charlie Puth   
3                                                 SNAP         Rosa Linn   
4                                              Sunroof       Nicky Youre   
..                                                 ...               ...   
955                                                Oh!  The Linda Lindas   
956                                 God Complex (Mojo)         poutyface   
957                                            I EXIST            VUKOVI   
958                       It Took A Lot To Get To This      Dylan Fraser   
959                                      Spirit Rising      Philip Sayce   

                  Artist_ID  \
0    5Pwc4xIPtQLFEnJriah9YJ   
1    6KImCVD70vtIoJWnq6nGn3   
2    6VuMaDnrHyPL1p4EHjYLi7   
3    46xBNx0j6cwY6sD9LgMTm1   
4    7qmpXeNz2ojlMl2EEfkeLs   
..                      ...   
955  13dTrWNNrnZ3AkgNyQNKP5   
956  0H44O4IYqpTOGx4c5nV37f   
957  1844Ua6R4gOuH6GLdlR4dt   
958  6Awp6fgyzqQ2XuEvOMjq8D   
959  5Npr4HpRE8YlsisRjN9T8h   

                                          Artist_Genre  Duration_ms  \
0                    ['piano rock', 'pop', 'pop rock']       148485   
1                                              ['pop']       167303   
2                    ['dance pop', 'pop', 'viral pop']       154486   
3                                                   []       179551   
4                                                   []       163025   
..                                                 ...          ...   
955                                     ['riot grrrl']       155821   
956         ['alt z', 'modern indie pop', 'viral rap']       106740   
957  ['modern alternative rock', 'modern rock', 'pi...       269666   
958                                                 []       247545   
959  ['blues rock', 'modern blues', 'modern blues r...       169287   

     Popularity  Explicit  Danceability  ...  Mode  Speechiness  Acousticness  \
0            97     False         0.704  ...     1       0.0475      0.082600   
1            96     False         0.520  ...     0       0.0557      0.342000   
2            93     False         0.881  ...     1       0.0324      0.619000   
3            93     False         0.565  ...     1       0.0638      0.107000   
4            93     False         0.768  ...     1       0.0401      0.352000   
..          ...       ...           ...  ...   ...          ...           ...   
955          48     False         0.639  ...     1       0.0609      0.000039   
956          48      True         0.573  ...     0       0.0571      0.004390   
957          49      True         0.469  ...     1       0.0680      0.000063   
958          41      True         0.469  ...     1       0.0577      0.075800   
959          44     False         0.400  ...     0       0.0479      0.009820   

     Instrumentalness  Liveness  Valence    Tempo            Pl

### Recode Explicit 

In [16]:
def explicit_recode(song):
    if song == False:
        return 0
    if song == True:
        return 1

In [17]:
master_no_dupes['ExplicitR']= master_no_dupes['Explicit'].apply(explicit_recode)
master_no_dupes

Name                Track_ID  \
0                            I Ain't Worried  4h9wh7iOZ0GGn8QVp4RAOB   
1                                  As It Was  4LRPiXqCikLlN15c3yImP7   
2    Left and Right (Feat. Jung Kook of BTS)  0mBP9X2gPCuapvpZ7TGDk3   
3                                       SNAP  76OGwb5RA9h4FxQPT33ekc   
4                                    Sunroof  4h4QlmocP3IuwYEj2j14p8   
..                                       ...                     ...   
955                                      Oh!  4e72LGraX8sW5iztJvS5fo   
956                       God Complex (Mojo)  1B3ouXA9iRkqY1ywvZbzZ5   
957                                  I EXIST  3eI62ZWWxkl4LUaBqXvA6Y   
958        Apartment Complex on the Eastside  1cRk9clJhi67mcfTxNfnjU   
959                      Black Roller Coming  5OLFlnmmC8gQHiFisYvrAh   

                                                 Album            Artist  \
0    I Ain’t Worried (Music From The Motion Picture...       OneRepublic   
1                                            As It Was      Harry Styles   
2              Left and Right (Feat. Jung Kook of BTS)      Charlie Puth   
3                                                 SNAP         Rosa Linn   
4                                              Sunroof       Nicky Youre   
..                                                 ...               ...   
955                                                Oh!  The Linda Lindas   
956                                 God Complex (Mojo)         poutyface   
957                                            I EXIST            VUKOVI   
958                       It Took A Lot To Get To This      Dylan Fraser   
959                                      Spirit Rising      Philip Sayce   

                  Artist_ID  \
0    5Pwc4xIPtQLFEnJriah9YJ   
1    6KImCVD70vtIoJWnq6nGn3   
2    6VuMaDnrHyPL1p4EHjYLi7   
3    46xBNx0j6cwY6sD9LgMTm1   
4    7qmpXeNz2ojlMl2EEfkeLs   
..                      ...   
955  13dTrWNNrnZ3AkgNyQNKP5   
956  0H44O4IYqpTOGx4c5nV37f   
957  1844Ua6R4gOuH6GLdlR4dt   
958  6Awp6fgyzqQ2XuEvOMjq8D   
959  5Npr4HpRE8YlsisRjN9T8h   

                                          Artist_Genre  Duration_ms  \
0                    ['piano rock', 'pop', 'pop rock']       148485   
1                                              ['pop']       167303   
2                    ['dance pop', 'pop', 'viral pop']       154486   
3                                                   []       179551   
4                                                   []       163025   
..                                                 ...          ...   
955                                     ['riot grrrl']       155821   
956         ['alt z', 'modern indie pop', 'viral rap']       106740   
957  ['modern alternative rock', 'modern rock', 'pi...       269666   
958                                                 []       247545   
959  ['blues rock', 'modern blues', 'modern blues r...       169287   

     Popularity  Explicit  Danceability  ...  Speechiness  Acousticness  \
0            97     False         0.704  ...       0.0475      0.082600   
1            96     False         0.520  ...       0.0557      0.342000   
2            93     False         0.881  ...       0.0324      0.619000   
3            93     False         0.565  ...       0.0638      0.107000   
4            93     False         0.768  ...       0.0401      0.352000   
..          ...       ...           ...  ...          ...           ...   
955          48     False         0.639  ...       0.0609      0.000039   
956          48      True         0.573  ...       0.0571      0.004390   
957          49      True         0.469  ...       0.0680      0.000063   
958          41      True         0.469  ...       0.0577      0.075800   
959          44     False         0.400  ...       0.0479      0.009820   

     Instrumentalness  Liveness  Valence    Tempo            Playlist  \
0            0.000745    0.0546    0.825  139.994        Mood_

### Accessing Individual genres from the lists of genres

In [18]:
genre_list = [eval(l) for l in list(master_no_dupes["Artist_Genre"])]

In [19]:
flat_list = list(set([genre for sublist in genre_list for genre in sublist]))

In [20]:
len(flat_list)

429

### I really wanted to use genres, but I just don't think the amount of work is worth what it will contribute. I mean, first off, it's not even the genre of the individual songs, but of the artist. Secondly, there are 429 genres across all of the songs. Some of them aren't even really genres, but just playlists on spotify. Genres of music are so detailed and a great classification tool for music. But I don't think classifying each song's genre or genres is really going to be helpful for this project. 

### Creating a csv file for the data wrangled dataset.

In [24]:
master_no_dupes.to_csv('master_mood.csv')